src/api/depositV2.test.js

```diff
@@ -0,0 +1,207 @@
+const nock = require('nock')
+const instance = require('./test/helpers/instance')
+const mockGetConf = require('./test/fixtures/getConf')
+
+let dvf
+let ethSendMock
+// Stubs the 'send' function for blockchain transactions and returns
+// choosen status and tx hash
+const stubDvfEthSend = (dvfEth, {stubTxHash, status = true}) => {
+  ethSendMock = jest.spyOn(dvfEth, 'send')
+    .mockImplementation(async (abi, address, action, args, value, options) => {
+      options.transactionHashCb(stubTxHash)
+      return {status}
+    })
+  return ethSendMock
+}
+
+describe('dvf.depositV2', () => {
+  beforeAll(async () => {
+    nock.cleanAll()
+    mockGetConf()
+    dvf = await instance()
+  })
+
+  beforeEach(() => {
+    nock.cleanAll()
+  })
+
+  afterEach(() => {
+    ethSendMock.mockReset()
+  })
+
+  afterAll(() => {
+    ethSendMock.mockRestore()
+  })
+
+  it(`Deposits ERC20 token to user's vault`, async () => {
+    const amount = '1394.0000051'
+    const token = 'USDT'
+
+    const stubTxHash = '0x123456'
+    const stubSendFn = stubDvfEthSend(dvf.eth, {stubTxHash})
+
+    const expectedPayload = {
+      token,
+      // Amount for USDT is quantised to 6 decimals, rounded down
+      amount: '1394000005',
+      txHash: stubTxHash
+    }
+
+    const apiResponse = {
+      ...expectedPayload,
+      pending: true,
+      expiresAt: (new Date()).toString(),
+      tx: {
+        vaultId: 1234,
+        tokenId: 5678,
+        starkKey: '0x',
+        amount: '1394000005'
+      }
+    }
+
+    const payloadValidator = jest.fn((body) => {
+      expect(body).toMatchObject(expectedPayload)
+      return true
+    })
+
+    nock(dvf.config.api)
+      .post('/v1/trading/deposits', payloadValidator)
+      .reply(200, apiResponse)
+
+    await dvf.depositV2({ token, amount })
+
+    expect(payloadValidator).toBeCalled()
+    expect(stubSendFn.mock.calls.length).toBe(1)
+  })
+
+  it('Deposits ETH to users vault', async () => {
+    mockGetConf()
+    const token = 'ETH'
+    const amount = '1.1177777777'
+
+    const stubTxHash = '0x123456'
+    const stubSendFn = stubDvfEthSend(dvf.eth, {stubTxHash})
+
+    const expectedPayload = {
+      token,
+      // Amount for ETH is quantised to 8 decimals, rounded down
+      amount: '111777777',
+      txHash: stubTxHash
+    }
+
+    const apiResponse = {
+      ...expectedPayload,
+      pending: true,
+      expiresAt: (new Date()).toString(),
+      tx: {
+        vaultId: 1234,
+        tokenId: 5678,
+        starkKey: '0x',
+        amount: '1394000005'
+      }
+    }
+
+    const payloadValidator = jest.fn((body) => {
+      expect(body).toMatchObject(expectedPayload)
+      return true
+    })
+
+    nock(dvf.config.api)
+      .post('/v1/trading/deposits', payloadValidator)
+      .reply(200, apiResponse)
+
+    await dvf.depositV2({ token, amount })
+
+    expect(payloadValidator).toBeCalled()
```

Comment:
for future reference, you can call do the following to assert that given nock mock has been executed:

```js
const nockScope = nock().post().reply()

...
// This will throw is given mock has not been matched yet
nockScope.done()
```

https://github.com/DeversiFi/launch-deversifi/blob/e3382b9540328d1f7cfe9fb12385f2f1259273a4/dvf-stark-trading/src/lib/stark/lastTxIdFromBatchId.test.js#L16-L21

there is also [`nock.isDone()`](https://github.com/nock/nock#isdone) (available on global or local scope) which returns a boolean and [`nock.pendingMocks`](https://github.com/nock/nock#pendingmocks) which returns an array

not that `.done()` is not available on global scope, but the latter 2 are